src/twisted/protocols/amp.py

```diff
@@ -2613,6 +2643,52 @@ def connectionLost(self, reason):
         self.transport = None
 
 
+class AMPv2(AMP):
+    """
+    This protocol implements the AMP version 2 protocol that allows values longer than 65535 bytes.
+    """
+
+    _currentValue = None  # type: Optional[List[bytes]]
+
+    def _write_box(self, box: AmpBox) -> None:
+        """Send an AmpBox using the AMPv2 format."""
+        assert self.transport is not None
+        self.transport.write(box.serialize(version2=True))
+
+    def proto_value(self, string: bytes) -> str:
```

Comment:
These methods do not belong on an `AMP` subclass, if they belong on a subclass at all, but a `BinaryBoxProtocol` subclass.

There's a mea-culpa here too; the inheritance-for-convenience design of AMP was a mistake, and should not have created this dichotomy; but, this is also why this should not be a new class, though, and instead just a parameter to `BinaryBoxProtocol` that `AMP` can propagate.

Reply:
Yeah, I got hit by that. My initial unpublished patch was version parameters everywhere.